File: Daily Challenge/March 2025/Minimum Recolors to Get K Consecutive Black Blocks.cpp
class Solution {
public:
  int minimumRecolors(string blocks, int k) {
    int n = blocks.size();
    vector<int> prefixSumOfWhite(n + 1, 0);
    for (int i = 0; i < n; i++) {
      prefixSumOfWhite[i + 1] = prefixSumOfWhite[i] + (blocks[i] == 'W');
    }

    int minOperations = INT_MAX;
    for (int i = 1; i + k - 1 <= n; i++) {
      int operations = prefixSumOfWhite[i + k - 1] - prefixSumOfWhite[i - 1];
      minOperations = min(minOperations, operations);
    }

    return minOperations;
  }
};

File: Daily Challenge/March 2025/Minimum Recolors to Get K Consecutive Black Blocks.cs
public class Solution {
  public int MinimumRecolors(string blocks, int k) {
    int n = blocks.Length;
    int[] prefixSumOfWhite = new int[n + 1];
    for (int i = 0; i < n; i++) {
      prefixSumOfWhite[i + 1] = prefixSumOfWhite[i] + (blocks[i] == 'W' ? 1 : 0);
    }

    int minOperations = int.MaxValue;
    for (int i = 1; i + k - 1 <= n; i++) {
      int operations = prefixSumOfWhite[i + k - 1] - prefixSumOfWhite[i - 1];
      minOperations = Math.Min(minOperations, operations);
    }

    return minOperations;
  }
}